hackathon_site/hardware/test_api.py

```diff
@@ -607,6 +608,58 @@ def test_search_filter(self):
         self.assertCountEqual(returned_ids, [self.order_3.id, self.order_4.id])
 
 
+class IncidentListViewPostTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.team = Team.objects.create()
+        self.order = Order.objects.create(status="Cart", team=self.team)
+        self.permissions = Permission.objects.filter(
+            content_type__app_label="hardware", codename="add_incident"
+        )
+
+        self.hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.order_item = OrderItem.objects.create(
+            order=self.order, hardware=self.hardware,
+        )
+
+        self.incident = Incident.objects.create(
+            state="Broken",
+            description="Description",
+            order_item=self.order_item,
+            time_occurred=datetime(2022, 8, 8, tzinfo=settings.TZ_INFO),
+        )
+
+        self.request_data = IncidentCreateSerializer(self.incident).data
+        self.incident.delete()
+
+        self.view = reverse("api:hardware:incident-list")
+
+    def test_user_not_logged_in(self):
+        response = self.client.post(self.view, self.request_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_incorrect_permissions(self):
+        self._login()
+        response = self.client.post(self.view, self.request_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_successful_post(self):
+        self._login(self.permissions)
+        self.profile = Profile.objects.create(user=self.user, team=self.team)
+        response = self.client.post(self.view, self.request_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
make an assertion on the response data as well

Reply:
yep will do